### PR TITLE
Convert coffeescript to es6

### PR DESCRIPTION
**Pull request by Katrienvh**
- Convert coffee script to ES6
- Remove coffee script gulp task and add task to compile ES6

### CODE REVIEW

File: src/SumoCoders/FrameworkCoreBundle/Resources/views/base.html.twig [hunk 1]
```diff
@@ -44,16 +44,7 @@
       <script type="text/javascript" src="{{ asset('/bundles/mopabootstrap/js/mopabootstrap-collection.js') }}"></script>
       <script type="text/javascript" src="{{ asset('/assets/js/sumo_plugins.js') }}"></script>
 
-      {% if app.environment == 'dev' %}
```
Comment (jonasdekeukelaere) [inline src/SumoCoders/FrameworkCoreBundle/Resources/views/base.html.twig]: so is there no method to add js files separately anymore?

Comment (jonasdekeukelaere) [inline src/SumoCoders/FrameworkCoreBundle/Resources/views/base.html.twig]: or is bundle.js not minified?

File: gulpfile.js [hunk 2]
```diff
@@ -56,14 +56,17 @@ var commonWebpackConfig = {
...
 gulp.task('webpack:generate-production-js', function() {
-  return gulp.src('src/**/Resources/assets/js/index.js')
+  return gulp.src([
+    'src/**/Resources/assets/js/index.js',
+    'vendor/sumocoders/**/Resources/assets/js/**'
+  ])
       .pipe(webpackStream(Object.assign({}, commonWebpackConfig, {
         plugins: [
-          new webpack.optimize.UglifyJsPlugin({
+          /*new webpack.optimize.UglifyJsPlugin({
```
Comment (jonasdekeukelaere) [inline gulpfile.js]: comment tags

File: src/SumoCoders/FrameworkCoreBundle/Resources/views/base.html.twig [hunk 3]
```diff
@@ -44,9 +44,6 @@
       <script type="text/javascript" src="{{ asset('/bundles/mopabootstrap/js/mopabootstrap-collection.js') }}"></script>
       <script type="text/javascript" src="{{ asset('/assets/js/sumo_plugins.js') }}"></script>
 
-      {% if bundleExists('SumoCodersFrameworkSearchBundle') %}
-        <script type="text/javascript" src="{{ asset('/assets/js/frameworksearchbundle.framework.form.search.js') }}"></script>
```
Comment (jonasdekeukelaere) [inline src/SumoCoders/FrameworkCoreBundle/Resources/views/base.html.twig]: how is this imported now?

File: src/SumoCoders/FrameworkCoreBundle/Resources/views/base.html.twig (hunk 1 above)
Comment (Katrienvh) [inline src/SumoCoders/FrameworkCoreBundle/Resources/views/base.html.twig]: bundle.js is generated with webpack and is minified. If you want to add a new file to bundle, you have to import it in index.js. Examples here: https://github.com/sumocoders/Framework/blob/convert-coffeescript-to-es6/src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/index.js

File: src/SumoCoders/FrameworkCoreBundle/Resources/views/base.html.twig (hunk 3 above)
Comment (Katrienvh) [inline src/SumoCoders/FrameworkCoreBundle/Resources/views/base.html.twig]: https://github.com/sumocoders/FrameworkSearchBundle/blob/convert-coffeescript-to-es6/Resources/assets/js/FrameworkSearch.js#L75

File: gulpfile.js (hunk 2 above)
Comment (Katrienvh) [inline gulpfile.js]: fixed 722371b

File: src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Locale.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
+import {Data} from 'Framework/Data';
+
+const dataData = new Data();
+
+export class Locale
+{
+  constructor()
+  {
+    this.isInitialized = false;
+    this.data = null;
```
Comment (Katrienvh) [inline src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Locale.js]: fixed f2149ff

File: src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Scrolling.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
+export class Scrolling
+{
+  constructor()
+  {
+    $('a.backToTop').on('click', $.proxy(this.scrollToTop, this));
```
Comment (Katrienvh) [inline src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Scrolling.js]: fixed 2e7983e

File: src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Slider.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+  initSlider()
+  {
+    this.element.slider(
+      {
+        min: 0,
+        max: 50,
+        values: [10, 40],
+        range: true,
+      }
```
Comment (Katrienvh) [inline src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Slider.js]: fixed b766115

File: src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Tabs.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+      /* an empty state object for now — either we implement a proper
+       popstate handler ourselves, or wait for jQuery UI upstream */
+      window.history.pushState({}, document.title, $current.href);
+    } else {
+      let scrolled = $(window).scrollTop();
+      window.location.hash = '#' + $current.href.split('#')[1];
+      $(window).scrollTop(scrolled);
+    }
+
+    $(this).tab('show');
```
Comment (Katrienvh) [inline src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Tabs.js]: fixed 3b30bcc

**Comment by jonasdekeukelaere** [comment]
@Katrienvh can you merge master in this branche to fix merge conflicts and tests?

**Pull request merged**